src/Amqp/JobQueue/AmqpWorker.php

```diff
@@ -195,19 +204,28 @@ private function recv(AMQPMessage $message)
         } catch (ErrorException $e) {
             $logLevel = LogLevel::ERROR;
             $logMessage = $this->handleFailure($message, $e, $logContext);
-        } catch (Exception $e) {
+        } catch (Throwable $e) {
+            if (null !== $this->errorHandler) {
+                try {
+                    $fn = $this->errorHandler;
+                    $fn($e);
+                } catch (Throwable $e) {
+                    $this->isStopping = true;
+                    $this->uncaughtThrowable = $e;
+                }
+            }
+
+            if ($e instanceof Error) {
```

Comment:
What does this do? It looks like it will hide the real exception from the log context?